app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -105,6 +109,7 @@
     public static final int STATE_UPGRADE = 1;

```

Comment:
Since these are now applicable mostly at either the AppRecord or CommCareApp level (IE: CommCare no longer can be in these states, only individual apps) It might make sense to move that flag to one of those methods.